domain/src/main/java/com/example/domain/base/BaseUseCase.kt

```diff
@@ -0,0 +1,7 @@
+package com.example.domain.base
+
+import com.example.domain.model.Result
+
+interface BaseUseCase<T : Any, R: Any> {
+  suspend operator fun invoke(param: T): Result<R>
+}
```

Comment:
Not sure if this class needed. it's not certain what all use cases will have only one invoke function

Reply:
Resolved